cmd/avalanche-network-runner/control/control.go

```diff
@@ -101,8 +109,25 @@ func startFunc(cmd *cobra.Command, args []string) error {
 	}
 	defer cli.Close()
 
+	opts := []client.OpOption{
+		client.WithNumNodes(numNodes),
+		client.WithPluginDir(pluginDir),
+	}
+	if customVMNameToGenesisPath != "" {
+		customVMs := make(map[string]string)
+		err = json.Unmarshal([]byte(customVMNameToGenesisPath), &customVMs)
+		if err != nil {
+			return err
+		}
+		opts = append(opts, client.WithCustomVMs(customVMs))
+	}
+
 	ctx, cancel := context.WithTimeout(context.Background(), requestTimeout)
-	info, err := cli.Start(ctx, avalancheGoBinPath, client.WithNumNodes(numNodes), client.WithWhitelistedSubnets(whitelistedSubnets))
+	info, err := cli.Start(
+		ctx,
+		avalancheGoBinPath,
+		opts...,
+	)
 	cancel()
```

Comment:
Since the change from @aaronbuchwald now propagates the `ctx` to all following function calls, and `Start` is an asynchronous call, we cannot cancel with `defer`. So, we can just set the timeout based on the request timeout, and discard the cancel function to set the top-level context.

Reply:
Added comments.